benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +43,187 @@ def get_data_model_from_task(task, param_dict):
                     f"Unsupported data_model '{data_model}' defined for task '{task}'"
                 )
     return None
+
+
+def _partition_input_files(input_files, task_options):
+    """
+    This function partitions the input files into witness and other files.
+    The distinction is based on the file name of the witness file, which
+    is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Tuple of witness files and other files
+    """
+    witness_files = []
+    other_files = []
+    for file in input_files:
+        if Path(file).name == task_options.options.get("witness"):
+            witness_files.append(file)
+        else:
+            other_files.append(file)
+    return witness_files, other_files
+
+
+def get_witness(task):
+    """
+    This function returns the unique witness file from the task.
+    It raises an exception if there are multiple witness files.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Unique witness file
+    """
+    witness_files, _ = _partition_input_files(
+        task.input_files_or_identifier, task.options
+    )
+    if len(witness_files) > 1:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks with more than one witness file"
```

Comment:
Thanks a lot for pointing this out!

Since currently the proposed format for validation tasks supports only a single witness, this is not yet relevant IMO. For an extension of the validation tasks to multiple witnesses, this definitely needs to be extended, but I would do this once it is relevant.

To make it easier to locate the the reason for a possible failure in the future I added a more explicit warning as you suggested in commit b864071578082f1b7c7c98eb3c877688dc4f4d9a